core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java

```diff
@@ -784,41 +807,49 @@ public InodeLockList lockDescendant(LockedInodePath inodePath, LockMode lockMode
    * @param inodePath the root {@link LockedInodePath} to retrieve all descendants from
    * @param lockMode the lock type to use
    * @return an {@link InodeLockList} representing the list of all descendants
-   * @throws FileDoesNotExistException if inode does not exist
    */
-  public InodeLockList lockDescendants(LockedInodePath inodePath, LockMode lockMode)
-      throws FileDoesNotExistException {
-    Inode<?> inode = inodePath.getInode();
-    InodeLockList inodeGroup = new InodeLockList();
-    if (!inode.isDirectory()) {
-      return inodeGroup;
-    }
-    return lockDescendantsInternal((InodeDirectory) inode, lockMode, inodeGroup);
+  public List<LockedInodePath> lockDescendants(LockedInodePath inodePath, LockMode lockMode) {
+    List<LockedInodePath> inodePathList = new ArrayList<>();
+    lockDescendantsInternal(inodePath, lockMode, inodePathList);
+    return inodePathList;
   }
 
-  private InodeLockList lockDescendantsInternal(InodeDirectory inodeDirectory,
-      LockMode lockMode, InodeLockList inodeGroup) {
+  private void lockDescendantsInternal(LockedInodePath inodePath, LockMode lockMode,
+                                      List<LockedInodePath> inodePathList) {
+    Inode<?> inode = inodePath.getInodeOrNull();
+    if (inode == null || !inode.isDirectory()) {
+      return;
+    }
+    InodeDirectory inodeDirectory = (InodeDirectory) inode;
     for (Inode<?> child : inodeDirectory.getChildren()) {
+      LockedInodePath lockedDescendantPath;
+      InodeLockList lockList = new InodeLockList();
       if (lockMode == LockMode.READ) {
         try {
-          inodeGroup.lockReadAndCheckParent(child, inodeDirectory);
+          lockList.lockReadAndCheckParent(child, inodeDirectory);
+          lockedDescendantPath = new MutableLockedInodePath(
```

Comment:
Can this functionality be provided by `LockedInodePath`? A `LockedInodePath` currently provides a new `LockedInodePath` for a non-existing child. There can be a similar method for returning a `LockedInodePath` of an existing child?

Reply:
done